tests/issue/issue.cpp

```diff
@@ -616,3 +626,697 @@ TEST(Issue, undefinedIssueUrl)
+    issue->setUnit(empty);
+    EXPECT_EQ(empty.first, issue->unit().first);
+    EXPECT_EQ(empty.second, issue->unit().second);
+}
+
+TEST(Issue, getMismatchedTypeReturnsNullUnits)
+{
+    auto units = libcellml::Units::create();
+    auto issue = libcellml::Issue::create();
+    issue->setUnits(units);
+
+    EXPECT_EQ(nullptr, issue->component());
+    EXPECT_EQ(nullptr, issue->componentRef());
+    EXPECT_EQ(nullptr, issue->connection().first);
+    EXPECT_EQ(nullptr, issue->encapsulation());
+    EXPECT_EQ(nullptr, issue->importSource());
+    EXPECT_EQ(nullptr, issue->mapVariables().first);
+    EXPECT_EQ(nullptr, issue->math());
+    EXPECT_EQ(nullptr, issue->model());
+    EXPECT_EQ(nullptr, issue->reset());
+    EXPECT_EQ(nullptr, issue->resetValue());
+    EXPECT_EQ(nullptr, issue->testValue());
+    EXPECT_EQ(nullptr, issue->unit().first);
+    EXPECT_NE(nullptr, issue->units());
+    EXPECT_EQ(nullptr, issue->variable());
+
+    issue->setUnits(nullptr);
+    EXPECT_EQ(nullptr, issue->units());
+}
+
+TEST(Issue, getMismatchedTypeReturnsNullVariable)
+{
+    auto variable = libcellml::Variable::create("variable");
+    auto issue = libcellml::Issue::create();
+    issue->setVariable(variable);
+    EXPECT_EQ(nullptr, issue->component());
+    EXPECT_EQ(nullptr, issue->componentRef());
+    EXPECT_EQ(nullptr, issue->connection().first);
+    EXPECT_EQ(nullptr, issue->encapsulation());
+    EXPECT_EQ(nullptr, issue->importSource());
+    EXPECT_EQ(nullptr, issue->mapVariables().first);
+    EXPECT_EQ(nullptr, issue->math());
+    EXPECT_EQ(nullptr, issue->model());
+    EXPECT_EQ(nullptr, issue->reset());
+    EXPECT_EQ(nullptr, issue->resetValue());
+    EXPECT_EQ(nullptr, issue->testValue());
+    EXPECT_EQ(nullptr, issue->unit().first);
+    EXPECT_EQ(nullptr, issue->units());
+    EXPECT_NE(nullptr, issue->variable());
+
+    issue->setVariable(nullptr);
+    EXPECT_EQ(nullptr, issue->variable());
+}
+
+TEST(Issue, clearStoredItem)
+{
+    auto model = libcellml::Model::create();
+    auto issue = libcellml::Issue::create(model);
+    EXPECT_EQ(model, issue->model());
+    EXPECT_EQ(libcellml::ItemType::MODEL, issue->itemType());
+
+    // Clear by setting to nullptr.
+    issue->setModel(nullptr);
+    EXPECT_EQ(nullptr, issue->model());
+    EXPECT_EQ(libcellml::ItemType::UNDEFINED, issue->itemType());
+
+    issue->setModel(model);
+    EXPECT_EQ(model, issue->model());
+    EXPECT_EQ(libcellml::ItemType::MODEL, issue->itemType());
+
+    // Clear by calling the clear function.
+    issue->clear();
+    EXPECT_EQ(nullptr, issue->model());
+    EXPECT_EQ(libcellml::ItemType::UNDEFINED, issue->itemType());
+}
+
+TEST(Issue, cannotCrossClearStoredItem)
+{
+    auto model = libcellml::Model::create();
+    auto issue = libcellml::Issue::create(model);
+    EXPECT_EQ(model, issue->model());
+    EXPECT_EQ(libcellml::ItemType::MODEL, issue->itemType());
+
+    // Cannot clear by setting to nullptr through another type.
```

Comment:
this is the behviour I am not sure I understand. Why does using a non-null object override whatever is there but a null object will respect the existing type. (Hoping this isn't opposite to anything I've said earlier!)

Reply:
I'll have another look at that.